src/System.Management.Automation/engine/SessionStateScope.cs

```diff
@@ -200,20 +150,6 @@ internal void NewDrive(PSDriveInfo newDrive)
                 throw e;
             }
 
-            if (!newDrive.IsAutoMounted &&
-                driveInfos.Count > DriveCapacity.FastValue - 1)
-            {
-                SessionStateOverflowException e =
-                    new SessionStateOverflowException(
```

Comment:
It seems that `SessionStateOverflowException` only gets constructed and thrown from within `SessionStateScope.cs` -- there are 7 references to this constructor and all of them are in this file.
We probably have to keep the class `SessionStateOverflowException` since it's a public API, even though it's not used anymore. But there are a lot `/// <exception cref="SessionStateOverflowException">` in comments, which we probably should clean up somehow.